pipeline/update_db.py

```diff
@@ -23,25 +35,63 @@ def get_db_connection():
     return eng
 
 
-def update_table(data, table, eng):
+def update_games(data, eng):
+    with eng.begin() as conn:
+        for _, row in data.iterrows():
+            stmt = f"SELECT * FROM {GAMES_TABLE} WHERE id = {row['id']};"
+            game = conn.execute(stmt).first()
+            # if game not in DB, insert all values and move on
+            if not game:
+                values = tuple(row[GAME_COLS])
+                stmt = f"INSERT INTO {GAMES_TABLE} VALUES {values};"
+                conn.execute(stmt)
+                continue
+            # if postponed/suspended, flag as date changed and nullify date
+            if row["date_changed"]:
+                stmt = (
+                    f"UPDATE {GAMES_TABLE} "
+                    f"SET date_changed = True, date = null "
+                    f"WHERE id = {row['id']};"
+                )
+                conn.execute(stmt)
+            # for games already in DB, update date/week/doubleheader
+            dt, wk, dh = row["date"], row["week"], row["dh"]
+            stmt = (
+                f"UPDATE {GAMES_TABLE} "
+                f"SET date = '{dt}', week = {wk}, dh = {dh} "
+                f"WHERE id = {row['id']};"
+            )
+            conn.execute(stmt)
+
+
+def update_outcomes(data, eng):
     with eng.begin() as conn:
-        # create temporary table with new data
-        data.to_sql("tmp", conn, index=False, if_exists="replace")
-        # remove those rows from games
-        stmt = f"DELETE FROM {table} WHERE id IN (SELECT id FROM tmp);"
-        conn.execute(stmt)
-        # insert new data into games
-        data.to_sql(table, conn, index=False, if_exists="append")
-        conn.execute("DROP TABLE tmp;")
+        for _, row in data.iterrows():
+            stmt = f"SELECT * FROM {OUTCOMES_TABLE} WHERE id = '{row['id']}';"
```

Comment:
Mark and I briefly discussed this a couple weeks ago and went with a meaningful primary key ID in outcomes: it's `<game_id>-0` or <game_id>-1` for when the selection is the home team or away team, respectively. So the lookup on ID alone does work, but I think what you're suggesting feels safer. What do you think?

Reply:
Wow I totally missed that in my first look at the scrape_data code. That should work fine, I don't have a strong preference one way or the other.

If we do want to stick with the <game_id>-<home/away> format, it might be good to add a comment somewhere calling that out - generally an ID generation strategy should be explicitly declared.

Reply:
Honestly the scrape_data module probably shouldn't be responsible for outcome ID generation at all, we can parse it out here or in a constructor for the `Outcome` model (it can be determined based on the `Game` fields plus the `selection_name`)

Reply:
Oh dang I love that idea